Dockerfile.template

```diff
@@ -91,6 +91,10 @@ RUN set -eux; \
 	cd /usr/src/memcached; \
 	\
 	gnuArch="$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)"; \
+# https://github.com/memcached/memcached/issues/1220#issuecomment-2770251664: on arm32, we need to override the upstream alignment check (which fails to detect the need for alignment on arm32v6+ on our hardware for some reason, which then causes us to fail the tests 😭)
+	case "$gnuArch" in \
+		arm-*abihf) export ac_cv_c_alignment=need ;; \
```

Comment:
The two values we need to capture here are `arm-linux-gnueabihf` and `arm-linux-musleabihf` (vs `arm-linux-gnueabi` which we *not* want to match).